libvast/include/vast/concept/printable/std/chrono.hpp

```diff
@@ -72,27 +74,43 @@ struct duration_printer : printer_base<duration_printer<Rep, Period, Policy>> {
 
 #undef UNIT_SUFFIX
 
+private:
+  static bool
+  print_adaptive(auto& out, double duration, std::string_view suffix) {
+    auto decimal = double{0.0};
+    const auto fractional = std::modf(duration, &decimal);
+    if (fractional == 0.0) {
+      fmt::format_to(out, "{:.1f}{}", duration, suffix);
+      return true;
+    }
+    auto double_str = fmt::format("{:.2f}", duration);
+    if (double_str.back() == '0')
+      double_str.pop_back();
```

Comment:
If I understand correctly then this works because the `:.2f` format specifier guarantees that there are exactly two trailing zeroes, so we must never remove more than one.

Do we then need the `fractional == 0.0` section above? If I understand correctly then L80-85 are not necessary in this function.

Reply:
It is not needed. Thanks for pointing this out.